### ISSUE TITLE
SHS - Wrong closure of headword at bracket

### ISSUE THREAD

**Issue by drdhaval2785** [sanskrit-lexicon/Cologne#181]
Please see
![capture](https://user-images.githubusercontent.com/5755132/30069670-feffc712-927e-11e7-918f-f2e3a80e9b11.png)

Here the headwords are erroneously stopped at nirDa only.

Actually it should be nirDana.

This is true for all of brackets of this dictionaries.

![capture](https://user-images.githubusercontent.com/5755132/30069738-2390f39e-927f-11e7-8d97-5b35a428caac.png)

Some kind of bug @funderburkjim ?

**Comment by gasyoun** [comment]
> This is true for all of brackets of this dictionaries.

Good catch, Dhaval.

**Comment by funderburkjim** [comment]
> nirDa error

Yep. This is an error in interpretation of parenthetical expressions within key2 for shs.
In shshw0, there are 839 entries  (out of 47310 total entries) which have a parenthesis.

The erroneous code is present in hw1.py removes from '(' to end of key2 in order to get key1:
`hw = re.sub('\(.*$','',hw)  # this is surely wrong for some.`

Now that you have shone a light on some where this is wrong, it is time to distinguish which are
indeed wrong.

In some of the 839, this procedure is right, like key2 = `aNgaja (-jaH)` -> key1 = `aNgaja`.

And for many roots the original procedure is also right: `kaba(f)kabf` -> key1 = `kaba`.

So the trick will be to decide which is which out of the 839.

Have added this to [TODO list](https://trello.com/b/sc1v74qW/todo-list).

**Comment by gasyoun** [comment]
https://trello.com/c/bqKD4nHp/49-shs-parentheses-in-key2-and-derivation-of-key1 great.

**Comment by gasyoun** [comment]
-

**Comment by gasyoun** [comment]
Fix applied in csl-orig PR: https://github.com/sanskrit-lexicon/csl-orig/pull/2870 — 8 k1 concat fixes + 2 k2 data errors corrected in v02/shs/shs.txt.
